polaris-react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -150,6 +152,7 @@ export function Tooltip({
       onMouseOver={handleMouseEnterFix}
       ref={setActivator}
       onKeyUp={handleKeyUp}
+      className={styles.TooltipContainer}
```

Comment:
I do not love adding this style directly to the Tooltip, however this does not appear to cause any breaking changes to the Tooltip directly, and fixes the header alignment.

I went with this approach since adding the flex property to the children within Tooltip, or the [IndexTable's header markup](https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/IndexTable/IndexTable.tsx#L916) caused the tooltip to align with the whole container:

<img width="294" alt="Screenshot 2023-01-12 at 4 20 59 PM" src="https://user-images.githubusercontent.com/22782157/212189894-7baa280d-e7f9-41a5-a674-e5a900fb48c5.png">

Any attempt to fix the Tooltip alignment regressed the original header issue, such as applying `inline-block` to the Tooltip's parent, and `flex` to the header markup. I.e, 

```html
<div style={{display: 'inline-block'}}>
   <span>
      <div style={{display: 'flex'}}>
          {headerMarkup}
      </div>
   </span>
</div>
```

Reply:
I think this is actually fixing a larger bug with the Tooltip! There was another person having a different issue earlier in the week, and when debugging his [codesandbox](https://codesandbox.io/s/beautiful-easley-wumczm?file=/App.js:65-92) the alignment was an extra bug that showed up.

My recommendation would just be to make a Playground that sets `Tooltip.activatorWrapper` to `div` (that prop is only used 4 times in Shopify/web and their each setting it to `div`). I'm guessing that won't have regressions, but just want to be sure before shipping this 💯

Reply:
I checked 3 cases of `activatorWrapper`'s that pass in a `div` and they look okay!
- InventoryList On hand tooltip
- Inventory On hand error icon
- Product category accept/reject buttons

<img width="365" alt="Screenshot 2023-01-13 at 2 19 43 PM" src="https://user-images.githubusercontent.com/22782157/212401175-350e0557-7766-474b-9ac5-f68f0c84c6e5.png">

Added a story for this:

<img width="250" alt="Screenshot 2023-01-13 at 2 26 32 PM" src="https://user-images.githubusercontent.com/22782157/212402331-41a1926b-db9e-43b4-8e94-075a4ed7b427.png">